src/Microsoft.SqlTools.ManagedBatchParser/Properties/AssemblyInfo.cs

```diff
@@ -34,11 +33,13 @@
 //      Build Number
 //      Revision
 //
-[assembly: AssemblyVersion("1.0.0.0")]
-[assembly: AssemblyFileVersion("1.0.0.0")]
-[assembly: InternalsVisibleTo("Microsoft.SqlTools.ServiceLayer.UnitTests")]
-[assembly: InternalsVisibleTo("Microsoft.SqlTools.ServiceLayer.IntegrationTests")]
-[assembly: InternalsVisibleTo("Microsoft.SqlTools.ServiceLayer.Test.Common")]
-[assembly: InternalsVisibleTo("MicrosoftSqlToolsServiceLayer")]
-[assembly: InternalsVisibleTo("MicrosoftKustoServiceLayer")]
-[assembly: InternalsVisibleTo("Microsoft.SqlTools.ManagedBatchParser.IntegrationTests")]
\ No newline at end of file
+[assembly: AssemblyVersion("3.0.0.0")]
+[assembly: AssemblyFileVersion("3.0.0.0")]
+#if NET6_0_OR_GREATER
+[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("Microsoft.SqlTools.ServiceLayer.UnitTests")]
+[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("Microsoft.SqlTools.ServiceLayer.IntegrationTests")]
+[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("Microsoft.SqlTools.ServiceLayer.Test.Common")]
+[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("MicrosoftSqlToolsServiceLayer")]
+[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("MicrosoftKustoServiceLayer")]
+[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("Microsoft.SqlTools.ManagedBatchParser.IntegrationTests")]
```

Comment:
These kind of things should really be in the .csproj file as
  <InternalsVisibleTo>
so they'd be right there with your build logic (you can make it conditional so that the key may or may not be specified based on the target framework...

FWIW, I tried to do it in this project but it did not work for me. Normal .csproj files (outside of the STS solution) work just fine... so I don't really know what custom logic you guys have in your .targets/.props that is overriding the behavior...

Reply:
I hear you. It's a web of internal dependencies as of now, in every single project. 
It needs to be done separately.